Secure api endpoint resolve issue #22

Tbones41 opened the pull request.

Tbones41 committed:
Revert "Correct Postgres port error"

This reverts commit 5a62a417e0598bd407fceac97def31138824472c.

Tbones41 committed:
Added authentication to test files

also corrected errored port in test.py

Tbones41 committed:
Correct flake8 errors

Nkoli commented on app/tests/test_authenticate.py:
```diff
@@ -0,0 +1,35 @@
+from django.contrib.auth.models import User
```
Use the same import name as you used in the base.py file. Makes it easier for people to understand.

Nkoli commented on app/tests/test_authenticate.py:
```diff
@@ -0,0 +1,35 @@
+from .base import BaseTestCase
+
+
+class AuthenticationTestCase(BaseTestCase):
```
[Test class naming convention](https://docs.python.org/3/library/unittest.html#basic-example). Please refer to the other preexisting test files for guidance too.

Nkoli reviewed (commented):
Please remember to add new lines at the end of the files you worked and re-add the spacing between import statements, also group them properly. Here's [why](https://www.python.org/dev/peps/pep-0008/#imports)

Tbones41 committed:
Revert changes to port

changed port from 5433 to 5432 in
`settings/test.py` and `settings/local.py`

removed colorama from `requirements.txt`

Tbones41 committed:
Increase time limit on access token.

reduce time limit on refresh token.
apply spacing to group imports.
rename class in `test_authenticate.py`
according to standard.

Nkoli commented on backend/settings/base.py:
```diff
@@ -140,3 +142,10 @@
+
+SIMPLE_JWT = {
+    'ACCESS_TOKEN_LIFETIME': timedelta(minutes=60),
+    'REFRESH_TOKEN_LIFETIME': timedelta(minutes=5),
```
I think the refresh token should have a longer life span than the access token, especially since the access token is dependent on the refresh token. [Use as a guide](https://django-rest-framework-simplejwt.readthedocs.io/en/latest/settings.html)

Tbones41 committed:
Change lifespan of access and refresh tokens

access token - 8 hours
refresh token - 6 months

kosyfrances commented on app/tests/base.py:
```diff
@@ -1,13 +1,30 @@
+from django.contrib.auth.models import User as Users
```
How about 
```
from django.contrib.auth.models import User as DjangoUser 
```
? 
since it is a Django inbuilt user not multiple users.

kosyfrances commented on app/tests/test_authenticate.py:
```diff
@@ -0,0 +1,35 @@
+from django.contrib.auth.models import User as Users
```
same thought as specified above

kosyfrances commented on app/tests/test_authenticate.py:
```diff
@@ -0,0 +1,35 @@
+
+class TestAuthenticationCase(BaseTestCase):
+
+    def test_successful_superuser_creation(self):
```
I'm not sure we need this test case. Why are you testing your test setup? 🤔

kosyfrances commented on app/tests/test_authenticate.py:
```diff
@@ -0,0 +1,35 @@
+            response = self.client.post('/api/token/', {
+                'username': 'testusertwo',
+                'password': 'testusertwopassword'
+            })
```
Why not use the user from the base case setup? 🤔

Nkoli commented on requirements.txt:
```diff
@@ -8,23 +7,25 @@ coveralls==1.11.1
 entrypoints==0.3
 flake8==3.7.9
 gunicorn==20.0.4
+httpie==2.1.0
```
please remove this, we don't need to test post and get through the terminal when we are already using postman

Nkoli commented on requirements.txt:
```diff
@@ -8,23 +7,25 @@ coveralls==1.11.1
 psycopg2==2.8.4
 pycodestyle==2.5.0
 pyflakes==2.1.1
+Pygments==2.6.1
```
do we need this?

Nkoli commented on requirements.txt:
```diff
@@ -8,23 +7,25 @@ coveralls==1.11.1
 pycodestyle==2.5.0
 pyflakes==2.1.1
+Pygments==2.6.1
+PyJWT==1.7.1
```
same here, is this needed?

Tbones41 replied:
PyJWT is the package needed for JWT. I uninstalled it and I could not run tests locally again

Tbones41 committed:
Resolve naming and coverage issues

change import from `Users` to `DjangoUser`
attempt to increase coverage in test_authenticate.py

Tbones41 committed:
Make corrections to requirements.txt

uninstall unneccesary packages
correct import names

Tbones41 committed:
Delete superuser testcase in test_authenticate.py

re-add PyJWT to packages in requirements.txt

Tbones41 committed:
add simpleJWT to requirements.txt

Tbones41 committed:
Readd JWT authentication

rebase branch
readd missing authentication code

Nkoli commented on requirements.txt:
```diff
@@ -1,30 +1,30 @@
 coveralls==1.11.1
 dj-database-url==0.5.0
-Django==3.0.7
+Django==3.0.3
```
we bumped this up to 3.0.7 earlier. The new version is a fix for some vulnerabilities, so please change the version back to that, and also update your django version in your mentorpaired dedicated virtualenv.

Tbones41 committed:
Change Django version to 3.0.7

kosyfrances reviewed (approved):
LGTM if there are no more objections from others.
